include/exadg/incompressible_navier_stokes/spatial_discretization/operators/viscous_operator.h

```diff
@@ -54,6 +54,23 @@ class ViscousKernel
   typedef CellIntegrator<dim, dim, Number> IntegratorCell;
   typedef FaceIntegrator<dim, dim, Number> IntegratorFace;
 
+  ElementType
+  get_element_type(IntegratorFace & integrator) const
+  {
+    if(integrator.get_matrix_free()
+         .get_dof_handler()
+         .get_triangulation()
+         .all_reference_cells_are_simplex())
+      return ElementType::Simplex;
+    else if(integrator.get_matrix_free()
+              .get_dof_handler()
+              .get_triangulation()
+              .all_reference_cells_are_hyper_cube())
+      return ElementType::Hypercube;
+    else
+      AssertThrow(false, dealii::ExcMessage("Only hypercube or simplex elements are supported."));
+  }
```

Comment:
We now need to realize this in a way that code duplication is avoided. What is the reason why this function is duplicated? (As a minor comment, I would prefer bracket around the return-statements for better readability.)